src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            return (List<T>) new BulldozerProducer().get();
+        }
+        if (Excavator.class.equals(type)) {
+            return (List<T>) new ExcavatorProducer().get();
+        }
+        if (Truck.class.equals(type)) {
+            return (List<T>) new TruckProducer().get();
```

Comment:
No need to create your `Producer`s every time you call `getAll` method, introduce them as field variables and initialize in constructor